DB/src/main/java/io/deephaven/db/v2/sources/ByteAsBooleanColumnSource.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+ */
+
+package io.deephaven.db.v2.sources;
+
+import io.deephaven.db.util.BooleanUtils;
+import io.deephaven.db.v2.sources.chunk.Attributes.Values;
+import io.deephaven.db.v2.sources.chunk.*;
+import io.deephaven.db.v2.utils.OrderedKeys;
+import io.deephaven.util.QueryConstants;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Reinterpret result {@link ColumnSource} implementations that translates {@link byte} to {@code Boolean} values.
+ */
+@AbstractColumnSource.IsSerializable(value = true)
+public class ByteAsBooleanColumnSource extends AbstractColumnSource<Boolean> implements MutableColumnSourceGetDefaults.ForBoolean {
+
+    private final ColumnSource<Byte> alternateColumnSource;
+
+    public ByteAsBooleanColumnSource(ColumnSource<Byte> alternateColumnSource) {
+        super(Boolean.class);
+        this.alternateColumnSource = alternateColumnSource;
+    }
+
+    @Override
+    public Boolean get(long index) {
+        final byte byteValue = alternateColumnSource.getByte(index);
+        if (byteValue == QueryConstants.NULL_BYTE) {
```

Comment:
This might merit a comment. Took me a minute to think through the possibilities, but I agree that this seems necessary given that `ReadOnlyIndex.NULL_KEY` should map to `NULL_BYTE` .

Reply:
We could also take the opportunity to fix the definition of NULL_BOOLEAN_AS_BYTE to be identical to NULL_BYTE since we do not have any legacy to care about.

Reply:
I think that can be done. I need to think a little bit about it on the Parquet side. This merits a ticket of its own.

Reply:
https://github.com/deephaven/deephaven-core/issues/927